### DISCUSSION TITLE
Microservices with Azure Container Apps and Bicep templates

### DISCUSSION THREAD

**Discussion by giscus** [CamiloTerevinto/Blog#9] [Announcements]
# Microservices with Azure Container Apps and Bicep templates

A look into how to deploy microservices to Azure Container Apps using Azure DevOps Pipelines for CI/CD and Azure Bicep templates.

https://www.camiloterevinto.com/post/microservices-container-apps-and-bicep

**Comment by NikolayBieliashov** [comment]
Hey! Thank you very much for your article! This is a great instruction!
I want to ask you one thing. How CA inside environment are talking to each other? Could Public API on your scheme runs a Traefic, for example, to play a role of reverse proxy for all microservices inside.

**Reply by CamiloTerevinto** [reply]
Hey @NikolayBieliashov, I'm glad the article was useful for you! 

You have a few options for the public API:

* You can use a reverse proxy container
* You can use the API aggregator pattern
* You can use the Backend For Frontend pattern

I personally use it as the entry point for validations, API documentation, and HTTP calls to other domain APIs

**Reply by NikolayBieliashov** [reply]
Thank you! I had some concerns about reverse proxy container. Try to test it.

**Reply by NikolayBieliashov** [reply]
@CamiloTerevinto sorry for stupid question, but I'm a bit confused and didn't find a lot of information about this.
Can I deploy LB container, opened to world, get a https-traffic on it, make tls offloading , and balance between internal containers?
Sounds to hard for CA, mb I should use external appGateway or smth?

**Reply by CamiloTerevinto** [reply]
@NikolayBieliashov I am not entirely sure why you'd want to do that though. CA already gives you load balancing and HTTPs with TLS offloading. If you can explain a bit more why that is not good enough, I might be able to help :)

**Reply by NikolayBieliashov** [reply]
Thank you for answer! I want to run three APIs inside CAenvironment which will be closed to world and one Traefik container that will be play a reverse proxy role. Connect my DNS name example.com (for example :)) and set up path-based routing: example.com/api1, example.com/api2 and example.com/api3.

**Reply by CamiloTerevinto** [reply]
@NikolayBieliashov I have never worked with Traefik to be honest with you, but this Microsoft sample suggests it's doable: https://github.com/Azure/reddog-containerapps

**Reply by NikolayBieliashov** [reply]
I really appreciate your participation! thank you and good luck!
